src/Core/Managed/Shared/Extensibility/TelemetryConfiguration.cs

```diff
@@ -313,6 +314,43 @@ public void Dispose()
         }
 
         /// <summary>
+        /// Adds a new telemetry sink to the configuration.
+        /// </summary>
+        /// <param name="sink">Sink to add.</param>
+        public void AddSink(TelemetrySink sink)
```

Comment:
I was under assumption that any pipeline that can be created via configuration should also be buildable programmatically.

I did not want to expose the TelemetrySinks collection directly because of the invariant that the first sink is the "default" sink, always present.